core/os/os_linux.odin

```diff
@@ -547,12 +547,23 @@ close :: proc(fd: Handle) -> Errno {
 	return _get_errno(unix.sys_close(int(fd)))
 }
 
+// If you read or write more than `SSIZE_MAX` bytes, result is implementation defined (probably an error).
+// `SSIZE_MAX` is also implementation defined but usually the max of a `ssize_t` which is `max(int)` in Odin.
+// In practice a read/write call would probably never read/write these big buffers all at once,
```

Comment:
So we need to wrap the internal `read` functionality then.

because what if you do need to read/write more than 1 GiB at a time? e.g. a very large file. This isn't that uncommon.

Reply:
I did think about that and came to the conclusion that you would be calling this, checking the returned int that says how much has been read, and call it again if needed, giving the most amount of control. Which still leaves looped implementations like `read_at_least`, `read_full` and `read_entire_file` for the rest.

If we were to say that `read` does the looping, it would basically be the same as `read_full`.

Reply:
Very good point!